tests/getopt/main.c

```diff
@@ -3,6 +3,16 @@
 
 #include "getopt.h"
 
+/*
+ * The test suite does not explicitly check the exit code of this program;
+ * instead, we rely on comparing the output with good output to determine
+ * success or failure.
```

Comment:
This is true, but a bit misleading.  If `test_getopt` fails, `test_getopt_args.sh` will abort early, and so it won't print the final "done".  So unless `test_getopt` somehow prints that by accident, having it exit with the wrong status (whether an incorrect failure or an incorrect success) will result in the test script output not matching and the test being flagged as a failure.

... at least, that's the theory.  Doesn't it work?

Reply:
I agree that's how it works.  I thought I described it acceptably with "does not **explicitly** check the exit code... rely on **comparing** the output", but evidently not.

Fun aside: what do you expect to see here?
```
#!/bin/sh -e
func() {
	echo "begin func"
	false
	echo "after false, ret code: $?"
}
echo "before calling func()"
func || true
echo "after calling func()"
```

Reply:
```
before calling func()
begin func
after calling func()
```

I think?

Reply:
I'd think so, too.  :)  dash 0.5.8 disagrees.

```
$ sh func-err.sh 
before calling func()
begin func
after false, ret code: 1
after calling func()
$ ls -l /bin/sh
lrwxrwxrwx 1 root root 4 Feb  5  2017 /bin/sh -> dash
$ dpkg -l dash | grep dash
ii  dash           0.5.8-2.1ubuntu2 amd64        POSIX-compliant shell
```

Reply:
Huh, maybe -e doesn't apply inside functions?  News to me...

Reply:
Just confirmed, FreeBSD sh does the same thing.